cuqi/experimental/mcmc/_cwmh.py

```diff
@@ -67,72 +67,71 @@ class CWMHNew(ProposalBasedSamplerNew):
         samples = sampler.sample(2000).get_samples()
 
     """
-    def __init__(self, target: cuqi.density.Density, proposal=None, scale=1,
+
+    _STATE_KEYS = ProposalBasedSamplerNew._STATE_KEYS.union(['scale_temp'])
+
+    def __init__(self, target:cuqi.density.Density=None, proposal=None, scale=1,
                  initial_point=None, **kwargs):
         super().__init__(target, proposal=proposal, scale=scale,
                          initial_point=initial_point, **kwargs)
-
-        # set initial scale
-        self.scale = scale
-
-        # set initial acceptance rate
+        
+    def _initialize(self):
+        self.scale_temp = self.scale
         self._acc = [np.ones((self.dim))]
 
     @property
     def scale(self):
         """ Get the scale parameter. """
+        if isinstance(self._scale, Number):
+            self._scale = np.ones(self.dim)*self._scale
         return self._scale
 
     @scale.setter
     def scale(self, value):
         """ Set the scale parameter. """
-        if isinstance(value, Number):
-            self._scale = np.ones(self.dim)*value
-        elif isinstance(value, np.ndarray):
-            self._scale = value
-        self._scale_temp = self._scale.copy()
+        self._scale = value
+        if isinstance(value, np.ndarray):
+            self._scale_temp = value.copy()
+        else:
+            self._scale_temp = value
```

Comment:
can we use the property immediately?
```suggestion
            self.scale_temp = value.copy()
        else:
            self.scale_temp = value
```